windows/src/developer/TIKE/compile/CompileKeymanWeb.pas

```diff
@@ -2326,19 +2326,25 @@ function TCompileKeymanWeb.GetKeyboardModifierBitmask: string;
 begin
   FBitMask := 0;
   gp := fk.dpGroupArray;
-  for i := 0 to fk.cxGroupArray-1 do
+  if fk.cxGroupArray > 0 then
```

Comment:
This was a separate bug (cxGroupArray is unsigned so if no groups/rules in a group, -1 gives integer underflow. I don't want to change the sign of the variable without a lot more validation so this is quick easy fix.